crates/solidity/outputs/spec/build.rs

```diff
@@ -33,14 +33,14 @@ fn generate_language_ebnf(
     let ebnf_model = EbnfModel::build(language);
 
     for (section_index, section) in language.sections.iter().enumerate() {
-        writer.buffer.push_str(&format!(
+        writer.write_comment(format!(
             "(*\n * {section_index}. {section_title}:\n *)\n\n",
             section_index = section_index + 1,
             section_title = section.title,
         ));
 
         for (topic_index, topic) in section.topics.iter().enumerate() {
-            writer.buffer.push_str(&format!(
+            writer.write_comment(format!(
```

Comment:
Oh dear, okay, I'm dumb; I didn't put the two and two together here and was ultimately confused by what `SimpleEbnfWriter` does here - it really is just appending to the buffer. In which case, I guess, it's fine to keep the buffer manipulation but why do we need the `EbnfWriter` trait again? Is this going to be used for the later pass, when we rewrite the spec-gen? If so, maybe we could add a comment that it's only used temporarily?

Reply:
> The spec will use the other implementation to write actual <span> and <a> tags that are used for syntax highlighting and cross linking.

I missed that, sorry. Feel free to add a comment or not; I'm assuming we will work on this shortly, anyway.

Reply:
>  Is this going to be used for the later pass, when we rewrite the spec-gen? If so, maybe we could add a comment that it's only used temporarily?

Based on this, I will actually create a general `PlainEbnfWriter` and add comments to it, as we will want to reuse it to write doc comments in kinds.rs.jinja2, similar to what we used to have, but this will probably have to wait for #638 to complete first.